File: Problem_2.js
// ## Problem2 Queue Reconstruction by Height (https://leetcode.com/problems/queue-reconstruction-by-height/)

// Time Complexity :O(n*nlogn)
// Space Complexity :O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No


// Your code here along with comments explaining your approach

/**
 * @param {number[][]} people
 * @return {number[][]}
 */
var reconstructQueue = function (people) {
    if (people === null || people.length === 0)
        return people;

    // Sort in descending order of height
    people.sort((a, b) => {
        if (a[0] === b[0]) {
            return a[1] - b[1];
        }
        return b[0] - a[0];
    })
    let result = [];
    // Place people at index same as number of people that should be ahead of them
    for (let i = 0; i < people.length; i++) {
        let person = people[i];
        result.splice(person[1], 0, person)
    }
    return result
};

File: problem2.cpp
/*
Time complexity: O(n*m)
Space complexity: O(1)
Did this code successfully run on Leetcode : Yes
Any problem you faced while coding this : No
*/

class Solution {
public:
    static bool srt(vector<int>& a, vector<int>& b){
        if(a[0]!=b[0])return a[0]>b[0];
        return a[1]<b[1];
    }
    vector<vector<int>> reconstructQueue(vector<vector<int>>& p) {
        sort(p.begin(), p.end(), srt);
        for(int i=0;i<p.size();i++){
            int j = i;
            while(j!=p[j][1]){
                swap(p[j], p[j-1]);
                j--;
            }
        }
        return p;
    }
};